provider/vsphere/fixture_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package vsphere_test
+
+import (
+	"archive/tar"
+	"bytes"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/imagemetadata"
+	"github.com/juju/juju/provider/vsphere"
+	coretesting "github.com/juju/juju/testing"
+)
+
+type ProviderFixture struct {
+	testing.IsolationSuite
+	dialStub testing.Stub
+	client   *mockClient
+	provider environs.EnvironProvider
+}
+
+func (s *ProviderFixture) SetUpTest(c *gc.C) {
+	s.IsolationSuite.SetUpTest(c)
+	s.dialStub.ResetCalls()
+	s.client = &mockClient{}
+	s.provider = vsphere.NewEnvironProvider(newMockDialFunc(&s.dialStub, s.client))
+}
+
+type EnvironFixture struct {
+	ProviderFixture
+	imageServer *httptest.Server
+	env         environs.Environ
+}
+
+func (s *EnvironFixture) SetUpTest(c *gc.C) {
+	s.ProviderFixture.SetUpTest(c)
+
+	s.imageServer = serveImageMetadata()
+	s.AddCleanup(func(*gc.C) {
+		s.imageServer.Close()
+	})
+
+	env, err := s.provider.Open(environs.OpenParams{
+		Cloud: fakeCloudSpec(),
+		Config: fakeConfig(c, coretesting.Attrs{
+			"image-metadata-url": s.imageServer.URL,
+		}),
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	s.env = env
+
+	// Make sure we don't fall back to the public image sources.
+	s.PatchValue(&imagemetadata.DefaultUbuntuBaseURL, "")
+	s.PatchValue(&imagemetadata.DefaultJujuBaseURL, "")
+}
+
+func serveImageMetadata() *httptest.Server {
+	index := `{
+  "index": {
```

Comment:
the thought had crossed my mind, but I don't think it's worthwhile - this is just a blob of simplestreams. it doesn't need to be structured, because we're never going to manipulate it